### PR TITLE
ch4/ofi: add back gni capability set

### LINKED ISSUE #4720: mpich 3.4a3 with ch4:ofi:gni hangs at message size > 8192 on Cori
```
cookbg@cori04:~/src/mpich-3.4a3> module list -l
- Package -----------------------------+- Versions -+- Last mod. ------
Currently Loaded Modulefiles:
modules/3.2.11.4                                     2019/10/23 20:26:46
gcc/8.3.0                                            2019/03/19 19:57:48
craype-haswell                                       2020/06/19 19:00:22
craype-hugepages2M                                   2020/06/19 19:00:22
craype-network-aries                                 2020/06/19 19:00:22
craype/2.6.2                                         2020/06/19 19:00:22
cray-mpich/7.7.10                                    2020/06/19 18:11:49
cray-libsci/19.06.1                                  2020/06/19 18:10:58
udreg/2.3.2-7.0.1.1_3.31__g8175d3d.ari               2020/06/04 19:51:11
ugni/6.0.14.0-7.0.1.1_7.33__ge78e5b0.ar              2020/06/11 16:50:33
pmi/5.0.14                                           2020/06/19 17:41:13
dmapp/7.1.1-7.0.1.1_4.48__g38cf134.ari               2020/06/11 17:01:45
gni-headers/5.0.12.0-7.0.1.1_6.28__g3b1              2020/06/11 16:46:01
xpmem/2.2.20-7.0.1.1_4.10__g0475745.ari              2020/06/04 19:44:21
job/2.2.4-7.0.1.1_3.36__g36b56f4.ari                 2020/06/04 19:44:17
dvs/2.12_2.2.156-7.0.1.1_8.9__g5aab709e              2020/06/11 17:14:31
alps/6.6.58-7.0.1.1_6.4__g437d88db.ari               2020/06/11 16:57:55
rca/2.2.20-7.0.1.1_4.46__g8e3fb5b.ari                2020/06/11 16:56:30
atp/2.1.3                                            2020/06/19 17:41:20
PrgEnv-gnu/6.0.5                                     2020/06/19 19:12:05
```

```
cookbg@cori04:~/src/mpich-3.4a3> ./configure --prefix=$(realpath ~/opt/mpich34) --with-device=ch4:ofi:gni --enable-ugni-static --with-pm=none --with-pmi=cray CC=gcc CXX=g++ FC=gfortran
```

```
cookbg@cori04:~/src/osu-micro-benchmarks-5.6.3> export PATH=~/opt/mpich34/bin:$PATH
cookbg@cori04:~/src/osu-micro-benchmarks-5.6.3> which mpicc
/global/homes/c/cookbg/opt/mpich34/bin/mpicc
cookbg@cori04:~/src/osu-micro-benchmarks-5.6.3> mpicc -show
gcc -I/global/u1/c/cookbg/opt/mpich34/include -L/global/u1/c/cookbg/opt/mpich34/lib -Wl,-rpath -Wl,/global/u1/c/cookbg/opt/mpich34/lib -Wl,--enable-new-dtags -lmpi
cookbg@cori04:~/src/osu-micro-benchmarks-5.6.3> ./configure CC=mpicc CXX=mpic++ CFLAGS="-march=core-avx2" CXXFLAGS="-march=core-avx2"
```

```
cookbg@cori04:~/src/osu-micro-benchmarks-5.6.3> salloc -q interactive -C haswell -N 2 -t 30
salloc: Granted job allocation 32966309
salloc: Waiting for resource configuration
salloc: Nodes nid000[58-59] are ready for job
cookbg@nid00058:~/src/osu-micro-benchmarks-5.6.3> srun -N2 -n2 -c2 --cpu-bind=cores ./mpi/pt2pt/osu_latency
# OSU MPI Latency Test v5.6.3
# Size          Latency (us)
0                       1.59
1                       1.62
2                       1.63
4                       1.63
8                       1.63
16                      1.62
32                      1.63
64                      1.63
128                     1.65
256                     1.82
512                     1.73
1024                    2.07
2048                    2.44
4096                    3.27
8192                    4.91
# THIS IS WHERE IT HANGS
```

Some samples with `perf record -g --pid <pid>` of

## rank 0

```
# Samples: 27  of event 'cycles:ppp'
# Event count (approx.): 49577050092341
#
# Children      Self  Command      Shared Object       Symbol                          
# ........  ........  ...........  ..................  ................................
#
   100.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] __gnix_cq_sreadfrom.isra.0
            |
            ---__gnix_cq_sreadfrom.isra.0
               |          
               |--96.12%--_gnix_prog_progress
               |          |          
               |          |--4.44%--_gnix_nic_progress
               |          |          |          
               |          |           --4.44%--__nic_tx_progress
               |          |          
               |           --2.97%--GNI_CqTestEvent
               |          
                --3.88%--__pthread_rwlock_rdlock

    96.12%    88.71%  osu_latency  libmpi.so.0.0.0     [.] _gnix_prog_progress
            |          
            |--88.71%--__gnix_cq_sreadfrom.isra.0
            |          _gnix_prog_progress
            |          
             --7.41%--_gnix_prog_progress
                       |          
                       |--4.44%--_gnix_nic_progress
                       |          |          
                       |           --4.44%--__nic_tx_progress
                       |          
                        --2.97%--GNI_CqTestEvent

     4.44%     0.00%  osu_latency  libmpi.so.0.0.0     [.] _gnix_nic_progress
            |          
             --4.44%--_gnix_nic_progress
                       |          
                        --4.44%--__nic_tx_progress

     4.44%     4.44%  osu_latency  libmpi.so.0.0.0     [.] __nic_tx_progress
            |          
             --4.44%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       |          
                        --4.44%--_gnix_nic_progress
                                  __nic_tx_progress

     3.88%     3.88%  osu_latency  libpthread-2.26.so  [.] __pthread_rwlock_rdlock
            |
            ---__gnix_cq_sreadfrom.isra.0
               __pthread_rwlock_rdlock

     2.97%     2.97%  osu_latency  libmpi.so.0.0.0     [.] GNI_CqTestEvent
            |
            ---__gnix_cq_sreadfrom.isra.0
               _gnix_prog_progress
               GNI_CqTestEvent

     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] GNII_DlaProgress
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] __nic_get_completed_txd
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] _gnix_prog_progress@plt
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] MPIDI_POSIX_eager_recv_begin
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] MPID_Progress_wait
     0.00%     0.00%  osu_latency  [unknown]           [.] 0x00093f7000000001
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] _gnix_vc_nic_progress
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] _gnix_cm_nic_progress
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] GNI_CqGetEvent.part.5
     0.00%     0.00%  osu_latency  [kernel]            [k] native_iret
     0.00%     0.00%  osu_latency  [unknown]           [k] 0xffffffff00000000
```

## rank 1

```
# Samples: 74K of event 'cycles:ppp'
# Event count (approx.): 64830618685
#
# Children      Self  Command      Shared Object       Symbol                          
# ........  ........  ...........  ..................  ................................
#
    50.78%     6.44%  osu_latency  libmpi.so.0.0.0     [.] __gnix_cq_sreadfrom.isra.0
            |
            ---__gnix_cq_sreadfrom.isra.0
               |          
               |--38.01%--_gnix_prog_progress
               |          |          
               |          |--25.26%--_gnix_nic_progress
               |          |          |          
               |          |          |--16.97%--__nic_tx_progress
               |          |          |          |          
               |          |          |          |--7.03%--__nic_get_completed_txd
               |          |          |          |          
               |          |          |          |--2.44%--GNI_CqGetEvent.part.5
               |          |          |          |          
               |          |          |           --0.63%--GNI_CqGetEvent@plt
               |          |          |          
               |          |          |--2.14%--_gnix_vc_nic_progress
               |          |          |          
               |          |          |--1.24%--pthread_spin_lock@plt
               |          |          |          
               |          |          |--0.87%--pthread_spin_unlock@plt
               |          |          |          
               |          |          |--0.62%--pthread_spin_lock
               |          |          |          
               |          |           --0.51%--__nic_get_completed_txd
               |          |          
               |          |--3.19%--GNI_CqTestEvent
               |          |          
               |          |--3.11%--_gnix_cm_nic_progress
               |          |          
               |           --0.57%--__nic_tx_progress
               |          
               |--3.13%--__pthread_rwlock_unlock
               |          
                --2.16%--__pthread_rwlock_rdlock

    38.14%     5.23%  osu_latency  libmpi.so.0.0.0     [.] _gnix_prog_progress
            |          
            |--33.06%--_gnix_prog_progress
            |          |          
            |          |--25.26%--_gnix_nic_progress
            |          |          |          
            |          |          |--16.97%--__nic_tx_progress
            |          |          |          |          
            |          |          |          |--7.03%--__nic_get_completed_txd
            |          |          |          |          
            |          |          |          |--2.44%--GNI_CqGetEvent.part.5
            |          |          |          |          
            |          |          |           --0.63%--GNI_CqGetEvent@plt
            |          |          |          
            |          |          |--2.14%--_gnix_vc_nic_progress
            |          |          |          
            |          |          |--1.24%--pthread_spin_lock@plt
            |          |          |          
            |          |          |--0.87%--pthread_spin_unlock@plt
            |          |          |          
            |          |          |--0.62%--pthread_spin_lock
            |          |          |          
            |          |           --0.51%--__nic_get_completed_txd
            |          |          
            |          |--3.19%--GNI_CqTestEvent
            |          |          
            |          |--3.11%--_gnix_cm_nic_progress
            |          |          
            |           --0.57%--__nic_tx_progress
            |          
             --5.09%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress

    25.63%     3.02%  osu_latency  libmpi.so.0.0.0     [.] _gnix_nic_progress
            |          
            |--22.61%--_gnix_nic_progress
            |          |          
            |          |--16.97%--__nic_tx_progress
            |          |          |          
            |          |          |--7.03%--__nic_get_completed_txd
            |          |          |          
            |          |          |--2.44%--GNI_CqGetEvent.part.5
            |          |          |          
            |          |           --0.63%--GNI_CqGetEvent@plt
            |          |          
            |          |--2.14%--_gnix_vc_nic_progress
            |          |          
            |          |--1.24%--pthread_spin_lock@plt
            |          |          
            |          |--0.87%--pthread_spin_unlock@plt
            |          |          
            |          |--0.62%--pthread_spin_lock
            |          |          
            |           --0.51%--__nic_get_completed_txd
            |          
             --3.02%--__gnix_cq_sreadfrom.isra.0
                       |          
                        --2.65%--_gnix_prog_progress
                                  _gnix_nic_progress

    17.53%     7.18%  osu_latency  libmpi.so.0.0.0     [.] __nic_tx_progress
            |          
            |--10.35%--__nic_tx_progress
            |          |          
            |          |--7.03%--__nic_get_completed_txd
            |          |          
            |          |--2.44%--GNI_CqGetEvent.part.5
            |          |          
            |           --0.63%--GNI_CqGetEvent@plt
            |          
             --7.18%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       |          
                       |--6.61%--_gnix_nic_progress
                       |          __nic_tx_progress
                       |          
                        --0.57%--__nic_tx_progress

    12.98%     0.00%  osu_latency  [unknown]           [.] 0000000000000000
            |
            ---0
               |          
               |--6.52%--GNI_CqGetEvent.part.5
               |          
               |--5.68%--GNI_CqGetEvent
               |          
                --0.74%--GNII_DlaProgress@plt

     8.96%     8.96%  osu_latency  libmpi.so.0.0.0     [.] GNI_CqGetEvent.part.5
            |          
            |--6.52%--0
            |          GNI_CqGetEvent.part.5
            |          
             --2.44%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       _gnix_nic_progress
                       __nic_tx_progress
                       GNI_CqGetEvent.part.5

     8.79%     8.78%  osu_latency  libmpi.so.0.0.0     [.] GNII_DlaProgress
            |          
             --8.76%--GNII_DlaProgress

     7.53%     7.53%  osu_latency  libmpi.so.0.0.0     [.] __nic_get_completed_txd
            |          
             --7.53%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       _gnix_nic_progress
                       |          
                       |--7.03%--__nic_tx_progress
                       |          __nic_get_completed_txd
                       |          
                        --0.51%--__nic_get_completed_txd

     6.90%     6.90%  osu_latency  libmpi.so.0.0.0     [.] progress_test
            |          
            |--3.93%--progress_test
            |          
             --2.97%--0x93f7000000001
                       progress_test

     5.94%     5.94%  osu_latency  libmpi.so.0.0.0     [.] GNI_CqGetEvent
            |          
             --5.68%--0
                       GNI_CqGetEvent

     5.20%     0.00%  osu_latency  [unknown]           [.] 0x00093f7000000001
            |
            ---0x93f7000000001
               |          
               |--2.97%--progress_test
               |          
                --1.75%--MPID_Progress_wait

     4.78%     4.77%  osu_latency  libmpi.so.0.0.0     [.] MPIDI_OFI_progress
            |
            ---MPIDI_OFI_progress

     3.72%     3.71%  osu_latency  libmpi.so.0.0.0     [.] MPIR_Progress_hook_exec_all
            |          
             --3.25%--MPIR_Progress_hook_exec_all

     3.54%     3.54%  osu_latency  libmpi.so.0.0.0     [.] MPIDI_POSIX_progress
            |
            ---MPIDI_POSIX_progress

     3.37%     3.36%  osu_latency  libpthread-2.26.so  [.] __pthread_rwlock_unlock
            |          
             --3.13%--__gnix_cq_sreadfrom.isra.0
                       __pthread_rwlock_unlock

     3.19%     3.19%  osu_latency  libmpi.so.0.0.0     [.] GNI_CqTestEvent
            |          
             --3.19%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       GNI_CqTestEvent

     3.15%     3.13%  osu_latency  libmpi.so.0.0.0     [.] _gnix_cm_nic_progress
            |          
             --3.13%--__gnix_cq_sreadfrom.isra.0
                       |          
                        --3.10%--_gnix_prog_progress
                                  _gnix_cm_nic_progress

     2.51%     2.51%  osu_latency  libmpi.so.0.0.0     [.] _gnix_vc_nic_progress
            |
            ---__gnix_cq_sreadfrom.isra.0
               _gnix_prog_progress
               |          
                --2.14%--_gnix_nic_progress
                          _gnix_vc_nic_progress

     2.21%     2.20%  osu_latency  libmpi.so.0.0.0     [.] MPIDI_POSIX_eager_recv_begin
            |
            ---MPIDI_POSIX_eager_recv_begin

     2.16%     2.16%  osu_latency  libpthread-2.26.so  [.] __pthread_rwlock_rdlock
            |
            ---__gnix_cq_sreadfrom.isra.0
               __pthread_rwlock_rdlock

     1.75%     1.75%  osu_latency  libmpi.so.0.0.0     [.] MPID_Progress_wait
            |
            ---0x93f7000000001
               MPID_Progress_wait

     1.37%     1.37%  osu_latency  libmpi.so.0.0.0     [.] pthread_spin_lock@plt
            |          
             --1.24%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       _gnix_nic_progress
                       pthread_spin_lock@plt

     1.19%     1.19%  osu_latency  libmpi.so.0.0.0     [.] pthread_spin_unlock@plt
            |          
             --0.86%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       _gnix_nic_progress
                       pthread_spin_unlock@plt

     0.98%     0.98%  osu_latency  libmpi.so.0.0.0     [.] MPIDI_SHM_progress
            |
            ---MPIDI_SHM_progress

     0.88%     0.88%  osu_latency  libmpi.so.0.0.0     [.] _gnix_queue_peek@plt
            |
            ---_gnix_queue_peek@plt

     0.85%     0.85%  osu_latency  libpthread-2.26.so  [.] pthread_spin_lock
            |          
             --0.62%--__gnix_cq_sreadfrom.isra.0
                       _gnix_prog_progress
                       _gnix_nic_progress
                       pthread_spin_lock

     0.74%     0.74%  osu_latency  libmpi.so.0.0.0     [.] GNII_DlaProgress@plt
            |
            ---0
               GNII_DlaProgress@plt

     0.63%     0.63%  osu_latency  libmpi.so.0.0.0     [.] GNI_CqGetEvent@plt
            |
            ---__gnix_cq_sreadfrom.isra.0
               _gnix_prog_progress
               _gnix_nic_progress
               __nic_tx_progress
               GNI_CqGetEvent@plt

     0.61%     0.61%  osu_latency  libmpi.so.0.0.0     [.] MPIDI_OFI_get_buffered
            |
            ---MPIDI_OFI_get_buffered

     0.46%     0.46%  osu_latency  libpthread-2.26.so  [.] pthread_spin_unlock
     0.44%     0.44%  osu_latency  libmpi.so.0.0.0     [.] _gnix_prog_progress@plt
     0.35%     0.35%  osu_latency  libmpi.so.0.0.0     [.] pthread_rwlock_unlock@plt
     0.32%     0.32%  osu_latency  libmpi.so.0.0.0     [.] _gnix_queue_peek
     0.28%     0.28%  osu_latency  libmpi.so.0.0.0     [.] pthread_rwlock_rdlock@plt
     0.22%     0.22%  osu_latency  libmpi.so.0.0.0     [.] GNI_CqTestEvent@plt
     0.19%     0.19%  osu_latency  libmpi.so.0.0.0     [.] _gnix_vc_nic_progress@plt
     0.12%     0.12%  osu_latency  libmpi.so.0.0.0     [.] gnix_cq_read
     0.06%     0.06%  osu_latency  [kernel]            [k] native_iret
     0.01%     0.01%  osu_latency  libmpi.so.0.0.0     [.] GNI_PostDataProbeById
     0.01%     0.01%  osu_latency  libmpi.so.0.0.0     [.] _gnix_dgram_poll
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] _gnix_dgram_poll@plt
     0.00%     0.00%  osu_latency  libmpi.so.0.0.0     [.] GNI_PostDataProbeById@plt
```

## separate libfabric

I also built my own libfabric 1.10.1 with 

```
./configure --prefix=$(realpath ~/opt/libfabric) CC=gcc --enable-gni=yes --enable-verbs=no --enable-ugni-static --enable-tcp=no --enable-usnic=no --enable-rxm=no --enable-mrail=no --enable-sockets=no --enable-rstream=no --enable-perf=no --enable-rxd=no --enable-udp=no --enable-hook_debug=no
```

MPICH 3.4a3 built against this has the same issue. If I instead use the verbs provider and verbs compatibility on Cray I get bad performance, but all message sizes work:

```
cookbg@nid00141:~/src/osu-micro-benchmarks-5.6.3> srun -n2 -N2 -c2 --cpu-bind=cores ./mpi/pt2pt/osu_latency
Unidentified node: Error detected by libibgni.so.  Subsequent operation may be unreliable.  IAA did not recognize this as an MPI process
Unidentified node: Error detected by libibgni.so.  Subsequent operation may be unreliable.  IAA did not recognize this as an MPI process
# OSU MPI Latency Test v5.6.3
# Size          Latency (us)
0                      10.69
1                      10.51
2                      10.80
4                      10.53
8                      10.58
16                     10.52
32                     10.51
64                     10.63
128                    10.69
256                    13.37
512                    15.69
1024                   19.08
2048                   18.51
4096                   18.70
8192                   19.75
16384                  20.74
32768                  23.97
65536                  38.41
131072                 50.26
262144                 73.14
524288                118.50
1048576               200.57
2097152               375.34
4194304               704.58
```

### libfabric only

The stand alone build of libfabric appears to work with expected performance.

`https://github.com/ofi-cray/cray-tests/blob/master/performance/multi-node/rdm_pingpong.c`

with `MAX_MSG_SIZE (1<<22)`

```
./configure --with-libfabric=$(realpath ~/opt/libfabric) --with-pmi=/opt/cray/pe/pmi/default CC=gcc
```

```
cookbg@cori04:~/src/cray-tests/performance/multi-node> srun -q interactive -C haswell -t 5 -N2 -n2 -c2 --cpu-bind=cores ./rdm_pingpong
1 threads
# Libfabric Latency Test
# Size            Latency (us)        Min Lat (us)        Max Lat (us)
1                         1.37                1.37                1.37
2                         1.35                1.35                1.35
4                         1.33                1.33                1.33
8                         1.33                1.33                1.33
16                        1.33                1.33                1.33
32                        1.33                1.33                1.33
64                        1.33                1.33                1.33
128                       1.34                1.34                1.34
256                       1.36                1.36                1.36
512                       1.39                1.39                1.39
1024                      1.61                1.61                1.61
2048                      1.91                1.91                1.91
4096                      2.46                2.46                2.46
8192                      3.70                3.70                3.70
16384                    13.98               13.98               13.98
32768                    12.95               12.95               12.95
65536                    17.87               17.87               17.87
131072                   24.78               24.78               24.78
262144                   36.77               36.77               36.77
524288                   69.20               69.20               69.20
1048576                 116.06              116.06              116.06
2097152                 225.39              225.39              225.39
4194304                 440.83              440.83              440.83
```

### PR DESCRIPTION
**Pull request by minsii**
## Pull Request Description

The PR is still WIP. Post here to refer to issue #4720 . By setting back the GNI capability set (specifically enable `FI_RMA` and `FI_TAGGED`), it may hide the bug that occurs at AM pipeline path for send/recv.

Fixes #4720

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

File: src/mpid/ch4/netmod/ofi/ofi_impl.h [hunk 1]
```diff
@@ -315,22 +315,31 @@ int MPIDI_OFI_pack_get(void *origin_addr, int origin_count,
...
-        /* An order size value of 0 indicates that ordering is not guaranteed. */
-        MPIR_Assert(MPIDI_OFI_global.max_order_war != 0);
-        max_size = MPL_MIN(max_size, MPIDI_OFI_global.max_order_war);
+    /* Check ordering limit:
+     * - A value of 0 indicates that ordering is not guaranteed, thus disable native atomics by return 0.
+     * - A value of -1 guarantees ordering for any data size, thus return the input size.
+     * - Otherwise return max_size based on the ordering limit. */
+    if (MPIDIG_WIN(win, info_args).accumulate_ordering & MPIDIG_ACCU_ORDER_WAR) {
+        if (MPIDI_OFI_global.max_order_war == 0) {
+            max_size = 0;
```
Comment (hzhou) [inline src/mpid/ch4/netmod/ofi/ofi_impl.h:324]: Just remove `MPIR_Assert`, `MPL_MIN` should work with `0`, right?

File: src/mpid/ch4/netmod/ofi/ofi_init.c [hunk 2]
```diff
@@ -1461,6 +1457,8 @@ static int find_provider(struct fi_info *hints)
 
         fi_freeinfo(prov_list);
     } else {
+        init_global_settings(MPIR_CVAR_OFI_USE_PROVIDER);
```
Comment (hzhou) [inline src/mpid/ch4/netmod/ofi/ofi_init.c]: While saving an extra branch, you are moving two code that belongs to the same logic (`init_global_settings`)  far apart, thus less than ideal.

Comment (minsii) [reply src/mpid/ch4/netmod/ofi/ofi_init.c]: I am unsure if I really want to cleanup the init code. The branch was created a while ago. Perhaps I will just drop the cleanup commits. However, even adding the capability back, pt2tp with >=16384bytes still hangs on cori. I need to investigate more.

Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/ofi_init.c]: What is the debug capability output after you apply the patch?

**Comment by minsii** [comment]
The PR now resolves the large PT2PT issue on Cori with a patch to ofi/gni provider [libfabric #6593](https://github.com/ofiwg/libfabric/pull/6593)

I also removed two commits which should be unnecessary to this PR.
- ch4/ofi: code format cleanup for find_provider routine
- ch4/ofi: fallback to AM-based RMA atomics if ordering is not provided

TODO:
- Run test suite on cori with this fix (we don't request failure free)

Future changes (will be addressed in separate PR)
- Run RMA tests and fix bugs
- Always register local mr and cache in AVL tree
    - Provider does not cache local MR thus explicitly caching in MPICH should be beneficial
- Try to enable `MPIDI_OFI_ENABLE_TAGGED_GNI` as `FI_TAGGED` is supported by gni now.
    -  This PR reverts the original capability set which sets TAGGED OFF.

**Comment by minsii** [comment]
test:mpich/ch4/ofi

**Comment by minsii** [comment]
@hzhou can you please help review this PR? It will take a while to run mpich test suite on cori.  I will report results later.

**Review by hzhou — APPROVED** [review]
Looks good. I am okay to merge it as long as our CI tests are clear and it resolves the osu latency tests on Cori. We always can do additional PRs for enable tags and fixing additional failures on Cori (if you find any).

**Comment by minsii** [comment]
test:mpich/ch4/ofi

**Pull request merged**